app/controllers/MovementDetailsController.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions.{AuthenticatedAction, JourneyRefiner}
+import controllers.exchanges.JourneyRequest
+import forms.MovementDetails._
+import forms.{ArrivalDetails, DepartureDetails}
+import javax.inject.{Inject, Singleton}
+import models.cache._
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents}
+import play.twirl.api.Html
+import repositories.MovementRepository
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.{arrival_details, departure_details}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class MovementDetailsController @Inject()(
+  authenticate: AuthenticatedAction,
+  getJourney: JourneyRefiner,
+  movementRepository: MovementRepository,
+  mcc: MessagesControllerComponents,
+  arrivalDetailsPage: arrival_details,
+  departureDetailsPage: departure_details
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
+
+  def displayPage(): Action[AnyContent] = (authenticate andThen getJourney).async { implicit request =>
+    val movementAnswers = request.answersAs[MovementAnswers]
+    movementAnswers.`type` match {
```

Comment:
You can do `getJourney(JourneyType.ARRIVE, JourneyType.DEPART)` to ensure the user is on the right journey